src/CategorySource.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\TranslatorExtractor;
+
+use RuntimeException;
+use Yiisoft\Translator\MessageReaderInterface;
+use Yiisoft\Translator\MessageWriterInterface;
+
+/**
+ * Represents message category for read and write messages
+ */
+final class CategorySource
+{
+    private string $name;
+    private MessageReaderInterface $reader;
+    private MessageWriterInterface $writer;
+
+    /**
+     * @param string $name Category name.
+     * @param MessageReaderInterface $reader
+     * @param MessageWriterInterface $writer
+     */
+    public function __construct(string $name, MessageReaderInterface $reader, MessageWriterInterface $writer)
+    {
+        if (!preg_match('/^[a-z0-9_-]+$/i', $name)) {
```

Comment:
Need extra test where category source name doesn't start with `[a-z0-9_-]`. For example, `!test`.

Reply:
Mutation in Uppercase (space also incorrect symbol)